lib/puppet/indirector/facts/facter.rb

```diff
@@ -32,8 +32,13 @@ def find(request)
     # Initialize core Puppet facts, such as puppetversion
     Puppet.initialize_facts
 
-    result = Puppet::Node::Facts.new(request.key, Facter.to_hash)
-    result.add_local_facts
+    result = if request.options[:show]
```

Comment:
I could imagine wanting to call `Puppet::Node::Facts.indirection.find` in other places besides the show action. Maybe call the option `:facter_options` or `:resolve_options`?

Reply:
I renamed the option to `:resolve_options`